kbase-extension/static/kbase/js/common/ui.js

```diff
@@ -335,120 +312,197 @@ define([
+     *                  with the value as the label
+     *      {string}    bsClass (opt) - the bootstrap theme to use for the modal
+     *                  one of 'primary', 'success', 'info', 'warning', 'danger'
+     *
+     * @returns {Promise} that resolves to false if the modal is dismissed using the
+     * cancel button, the close button, by clicking on the modal backdrop, or by pressing
+     * the escape key. It resolves to true if the ok button is clicked or the user hits
+     * enter.
+     */
+    function showGenericDialog(args) {
+        const dialog = _renderModal(args),
+            confirmNode = _generateConfirmNode(dialog),
+            modalDialogNode = _setUpModalNodes(confirmNode);
 
-        confirmNode.id = dialogId;
-        confirmNode.innerHTML = dialog;
+        // this shows the modal
+        $(modalDialogNode).modal({ keyboard: false });
 
-        // top level element for kbase usage
-        kbaseNode = document.querySelector('[data-element="kbase"]');
-        if (!kbaseNode) {
-            kbaseNode = document.createElement('div');
-            kbaseNode.setAttribute('data-element', 'kbase');
-            document.body.appendChild(kbaseNode);
-        }
+        let resolution = false;
 
-        // a node upon which to place Bootstrap modals.
-        modalNode = kbaseNode.querySelector('[data-element="modal"]');
-        if (!modalNode) {
-            modalNode = document.createElement('div');
-            modalNode.setAttribute('data-element', 'modal');
-            kbaseNode.appendChild(modalNode);
+        if (args.type === 'confirm') {
+            modalDialogNode.querySelector('[data-element="ok"]').addEventListener('click', () => {
+                resolution = true;
+            });
         }
 
-        modalNode.appendChild(confirmNode);
+        modalDialogNode.addEventListener('keyup', (e) => {
+            // 13 = enter, 27 = escape
+            // the key is sometimes a string, hence == instead of ===
+            if (e.key == 13 || e.key == 27) {
+                $(modalDialogNode).modal('hide');
+                if (e.key == 13 && args.type === 'confirm') {
+                    resolution = true;
+                }
+            }
+        });
 
-        const modalDialogNode = modalNode.querySelector('.modal');
-        $(modalDialogNode).modal('show');
         return new Promise((resolve) => {
-            modalDialogNode.querySelector('[data-element="ok"]').addEventListener('click', () => {
-                confirmNode.parentElement.removeChild(confirmNode);
-                resolve(false);
-            });
-            modalDialogNode.addEventListener('hide.bs.modal', () => {
-                resolve(false);
+            $(modalDialogNode).on('hidden.bs.modal', () => {
+                confirmNode.remove();
+                resolve(resolution);
             });
+            if (args.doThisFirst) {
+                args.doThisFirst();
+            }
```

Comment:
the infamous `doThisFirst` argument was necessary to make these modals testable.

Reply:
Can you add that to the commented args list?